Add siblings for respond assets.

cyber-pc opened the pull request:
This PR add support for respond asset with the siblings.
The example result is as below
```
./qubic-cli -nodeip <IP> -getasset IUSERMFNNDLCYBFJRARDDEZWNFOBDKBXXPZQYOZIAACDZEYQOOJPXEGDXKEE
======== OWNERSHIP ========
Asset issuer: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFXIB
Asset name: QX
Managing contract index: 1
Issuance Index: 1
Number Of Units: 68
Asset Digest: 46285cb31737544ed56c9b8ac9500bfec425e642d7e276a571f73088fb6a458c
Tick: 13820031
Asset issuer: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFXIB
Asset name: RANDOM
Managing contract index: 1
Issuance Index: 0
Number Of Units: 68
Asset Digest: 46285cb31737544ed56c9b8ac9500bfec425e642d7e276a571f73088fb6a458c
Tick: 13820031
======== POSSESSION ========
Asset issuer: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFXIB
Asset name: QX
Managing contract index: 1
Owner index: 13011080
Owner ID: IUSERMFNNDLCYBFJRARDDEZWNFOBDKBXXPZQYOZIAACDZEYQOOJPXEGDXKEE
Number Of Units: 68
Asset Digest: 46285cb31737544ed56c9b8ac9500bfec425e642d7e276a571f73088fb6a458c
Tick: 13820031
Asset issuer: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFXIB
Asset name: RANDOM
Managing contract index: 1
Owner index: 13011082
Owner ID: IUSERMFNNDLCYBFJRARDDEZWNFOBDKBXXPZQYOZIAACDZEYQOOJPXEGDXKEE
Number Of Units: 68
Asset Digest: 46285cb31737544ed56c9b8ac9500bfec425e642d7e276a571f73088fb6a458c
Tick: 13820031

####################################################
./qubic-cli -nodeip <IP> -getquorumtick ./computorlist.txt 13820031
Received 479 quorum tick #13820031 (votes)
Received 473 quorum tick #13820032 (votes)
Performing salt check...
ALL votes PASSED salts check
Number of unique votes: 1
Vote #0 (voted by 479 computors ID) Epoch: 108
Tick: 13820031
Time: 2024-05-08 12:20:44.0000
prevResourceTestingDigest: a99c15250fc377bb
prevSpectrumDigest: 804d96d95920db88fcb06cd6fae3ef65c295f71917e6ffdf76b90a885e7f0a20
prevUniverseDigest: 46285cb31737544ed56c9b8ac9500bfec425e642d7e276a571f73088fb6a458c
prevComputerDigest: a7cfad48b1151df81db03fc8b0c521de76d1f4e3d48528aff33f0b456c6be597
transactionDigest: d75fbd22f5ed6f67a08cfbb07972227579e406ea3303c354f46b8e4e47eded82
expectedNextTickTransactionDigest: 88e4b84fa018bd7337b6c118c6523424df3ca06bafc27e6edb81965db5b02ac6
```

krypdkat commented:
overall looks good to me, let get this merged on Epoch 110. After doing some testing on test-net

krypdkat commented on src/network_messages/common_def.h:
```diff
@@ -54,3 +57,17 @@ static inline bool operator!=(const IPv4Address& a, const IPv4Address& b)
+static void getSiblings(int digestIndex, const m256i* digests, m256i siblings[depth])
+{
+    const unsigned int capacity = (1ULL << depth);
+    int siblingIndex = digestIndex;
```
Do we need a lock here?
Same as the siblings for entity. Maybe while it updating `universe` & `spectrum`, these accesses would return wrong results. It's rare but it can happen.
please consider if we really need a lock here, (or users can just request again if the digest is mismatched).
If a lock is not needed, then maybe add some comments here about that race condition.

cyber-pc commented:
Thanks @krypdkat I addressed your comment.

I added a lock when calculating the the siblings, direst of spectrum in the 2nd commit. I think let the lock outside of this function for better control the resource.

For the assets, they've already had locks [here](https://github.com/qubic/core/blob/develop/src/assets.h#L329) and [here](https://github.com/qubic/core/blob/develop/src/assets.h#L365)
